src/tokenizer.rs

```diff
@@ -196,8 +196,25 @@ impl Tokenizer {
     return sentences;
   }
 
-  pub fn words<'a>(&self, sentence_range: &'a DNMRange) -> Vec<&'a str>  {
-    sentence_range.get_plaintext().split(|c: char| !c.is_alphabetic()).filter(|w| w.len() > 0).collect()
+  pub fn words<'a, 'b>(&'b self, sentence_range: &'a DNMRange<'b>) -> /* Vec<&'a str> */ Vec<DNMRange> {
```

Comment:
Thanks for taking the extra effort to make this "pure" and return a DNMRange, pretty cool.